File: jsParser/parser/ModificationVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Modification = require('../domain/Modification')

const Class_modificationVisitor = require('./Class_modificationVisitor')
const ExpressionVisitor = require('./ExpressionVisitor')

class ModificationVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitModification (ctx) {
    let class_modification = null
    const equal = !!ctx.SYMBOL_EQUAL()
    const colon_equal = !!ctx.SYMBOL_COLONEQUAL()
    let expression = null

    if (ctx.class_modification()) {
      const class_modificationVisitor = new Class_modificationVisitor.Class_modificationVisitor()
      class_modification = class_modificationVisitor.visitClass_modification(ctx.class_modification())
    }
    if (ctx.expression()) {
      const expressionVisitor = new ExpressionVisitor.ExpressionVisitor()
      expression = expressionVisitor.visitExpression(ctx.expression())
    }

    return new Modification.Modification(class_modification, equal, colon_equal, expression)
  }
};

ModificationVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitModification = this.visitModification
exports.ModificationVisitor = ModificationVisitor

File: java/parser/src/main/java/gov/lbl/parser/parser/ModificationVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Class_modification;
import gov.lbl.parser.domain.Expression;
import gov.lbl.parser.domain.Modification;

public class ModificationVisitor extends modelicaBaseVisitor<Modification> {
    @Override
    public Modification visitModification(modelicaParser.ModificationContext ctx) {    	  
        Class_modificationVisitor class_modificationVisitor = new Class_modificationVisitor();
        Class_modification class_modification = ctx.class_modification() == null ? null : ctx.class_modification().accept(class_modificationVisitor);

        ExpressionVisitor expressionVisitor = new ExpressionVisitor();
        Expression expression = ctx.expression() == null ? null : ctx.expression().accept(expressionVisitor);
        
        Boolean equal = ctx.SYMBOL_EQUAL() == null ? false : true;
        Boolean colon_equal = ctx.SYMBOL_COLONEQUAL() == null ? false : true;

        return new Modification(class_modification, equal, colon_equal, expression);
    }
}